src/entries/popup/pages/messages/useSimulateTransaction.tsx

```diff
@@ -125,6 +125,9 @@ export const useSimulateTransaction = ({
           domain,
         },
       )) as TransactionSimulationResponse;
+
+      if (!response?.simulateTransactions?.[0]) throw 'UNSUPPORTED';
```

Comment:
throwing the TransactionSimulationError type of `UNSUPPORTED` so that the response behaves as expected with the simulation overview etc (having an error but not a simulation response)